internal/app/machined/pkg/runtime/v1alpha1/platform/packet/packet.go

```diff
@@ -188,8 +189,20 @@ func (p *Packet) Mode() runtime.Mode {
 }
 
 // Hostname implements the platform.Platform interface.
-func (p *Packet) Hostname(context.Context) (hostname []byte, err error) {
-	return nil, nil
+func (p *Packet) Hostname(ctx context.Context) (hostname []byte, err error) {
+	log.Printf("fetching equinix metadata from: %q", PacketMetaDataEndpoint)
+
+	metadataConfig, err := download.Download(ctx, PacketMetaDataEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	var unmarshalledMetadataConfig Metadata
+	if err = json.Unmarshal(metadataConfig, &unmarshalledMetadataConfig); err != nil {
+		return nil, err
+	}
```

Comment:
Looks like we could optimize here a bit. We download the metadata twice. Not a big deal at all, but we have the `Packet` struct that we could store things to.